### DISCUSSION TITLE
BTT PAD 7 and the ADXL345

### DISCUSSION THREAD

**Discussion by Degenfoster** [bassamanator/Sovol-SV06-firmware#134] [Show and tell]
Hey sorry i'm quite new to klipper and i just got it to work with my sv06+ with your guide, but i can't find how to use the ADXL 345 for input shaping with the BTT pad7, enabling adxl direct caused an error and klipper doesn't launch, please forgive me I'm really beginner at coding and stuff so i don't really understand it

**Comment by MacBandito** [comment]
Did you follow the instructions in the Pad 7 manual?  You should have the BoardEnv.txt file that's installed when you follow the BTT instructions. You may have deleted it if you followed the bassamanator instructions. You need that file though to activate the correct setting. I’ve pasted the instructions from the manual below.  

#### Pad 7 Settings

Open the "BoardEnv.txt" configuration file, and set the following parameters (`spidev1_1` being the relevant bit):
```
overlays=ws2812 light mcp2515 spidev1_1
```

`spidev1_1`: Enables the spidev1_1 to the system user space, allowing the Pad7's SPI port to connect to an ADXL345 accelerometer module.

#### Setting up ADXL345

As explained in "Pad 7 Settings" section, set the overlays to include spidev1_1.
After booting, the system user space should load spidev1.1. Add the following
configuration to the printer.cfg file to use the ADXL345:

```
[mcu CB1]
serial: /tmp/klipper_host_mcu
```
```
[adxl345]
cs_pin: CB1:None
spi_bus: spidev1.1
axes_map: z,y,-x # Modify according to the actual orientation of the ADXL345
installed on the printer.
```

#### Sources

- [Official PAD7 Documentation](https://bigtreetech.github.io/docs/PAD7.html)
- [ADXL345 Section](https://bigtreetech.github.io/docs/PAD7.html#to-work-with-a-cm4)

**Comment by Degenfoster** [comment]
You are amazing! The config files of the pad 7 was already good, but the trick was the CB1 I kept following the wrong set of instructions for the CM4 board instead! So sorry rookie mistake but glad I finally could fix it!

**Comment by MacBandito** [comment]
You’re welcome.  I’m very new to this too.

**Comment by bassamanator** [comment]
@MacBandito Thanks for helping out the community! I've edited your original post for styling and to add sources. Please see if it still all makes sense or if I've blundered something.

**Comment by Taubenhaucher1969** [comment]
Hello everyone,
I converted my SV06+ with the Pad 7 to Klipper and it runs almost perfectly, the only problem I can't solve is the ADXL345.
First I followed the instructions from RosinDubh79, unfortunately without success.
Then I saw this thread, but I don't really understand it.
Since my Linux/Klipper knowledge is not that good I don't know where to place the "BoardEnv.txt".

My error message as soon as I include the [include ./cfgs/adxl-direct-pad7.cfg] is the following:

Klipper reports: ERROR

MCU Protocol error

This is frequently caused by running an older version of the
firmware on the MCU(s). Fix by recompiling and flashing the
firmware.

Your Klipper version is: v0.12.0-102-g9f41f53c
MCU(s) which should be updated:
mcu: Current version v0.11.0-148-g52f4e20c
CB1: Current version v0.11.0-122-ge6ef48cd
Up-to-date MCU(s):
<none>

Once the underlying issue is corrected, use the "RESTART"
command to reload the config and restart the host software.

mcu 'CB1': Command format mismatch: query_adxl345 oid=%c rest_ticks=%u vs query_adxl345 oid=%c clock=%u rest_ticks=%u

Can you help me with this?
Many thanks in advance.

**Reply by Rpm49** [reply]
I have this same issue.

This is what happens when we put the adxl information in it. @bassamanator any ideas? I updated the klipper.bin file to whatever 12 is now vs what is on this page. and still no luck. The CB1 i suspect is what the btt pad 7's board is on the back that runs it? I heard that someone re-flashed something with a new image for it or the pad. I might look at re-doing the firmware or something from scratch but I'm at a loss also @Taubenhaucher1969 Taubenhaucher1969

![image](https://github.com/bassamanator/Sovol-SV06-firmware/assets/123909250/175b343f-c999-4143-8f69-38aeb1364b10)

**Reply by bassamanator** [reply]
The CB1 firmware seems to be outdated enough to require an update?

**Reply by Degenfoster** [reply]
Hey! I received this message when i updated klipper a week ago, heres the steps to fix it!

Step 1: SSH into the btt pad 7 with a software like putty
Step 2: Log in, Username biqu, Password biqu
Step 3: type in cd klipper (not actually sure if this matters cause i do not know coding)
Step 4: type in "make menuconfig"
Step 5:  Change the settings as such
STM32F103
28KIB bootloader
Serial USART PA10/PA9
Disable SWD on start up - Checked
Then press Q and press Y when prompted to save
Step 6: type make
Step 7: use a software like WINSCP to go into /klipper/out and find "klipper.bin" copy into desktop, best to rename it like 1klipper cause it has to be a diffferent name from previous firmware
Step 8: transfer onto microSD card, (people say there are issues with the stock SV06+ card so play it safe and use something else
Step 9: Insert into printer when OFF, turn the printer on and wait a minute or 2, then turn off printer THEN remove the card
Step 10: In putty type make menuconfig
Step 11: Select controller to linux process, press q and save
Step 12: type Sudo service klipper stop, make flash and then Sudo service klipper start
Step 13: profit
(also make sure u go into the btt pad 7 settings and update the system if there is)
ORR just watch this video hahah
https://www.youtube.com/watch?v=cnthbR3tS9M&t=17s

**Reply by dasb00ter** [reply]
@Degenfoster If I klipperized my sv06 plus recently do I still need to flash the printer...or do I just need to update the "controller to linux process"

**Reply by Degenfoster** [reply]
@dasb00ter, only flash the printer if there is a version mismatch, Klipper will say "MCU version not compatible or something along those lines", happens when you update klipper

**Reply by dasb00ter** [reply]
This worked for me
I recompiled and did both printer and a pad flashing
